src/analyze/analyze-security.c

```diff
@@ -1527,7 +1608,109 @@ static const struct security_assessor security_assessor_table[] = {
         },
 };
 
-static int assess(const SecurityInfo *info, Table *overview_table, AnalyzeSecurityFlags flags, unsigned threshold) {
+static JsonVariant* security_assessor_find_in_policy(const struct security_assessor *a, JsonVariant *policy, const char *name) {
+        JsonVariant *item;
+        assert(a);
+
+        if (!policy)
+                return NULL;
+        if (!json_variant_is_object(policy)) {
+                log_debug("Specificied policy is not a JSON object, ignoring.");
+                return NULL;
+        }
+
+        item = json_variant_by_key(policy, a->json_field);
+        if (!item)
+                return NULL;
+        if (!json_variant_is_object(item)) {
+                log_debug("Item for '%s' in policy JSON object is not an object, ignoring.", a->id);
+                return NULL;
+        }
+
+        return name ? json_variant_by_key(item, name) : item;
+}
+
+static uint64_t access_weight(const struct security_assessor *a, JsonVariant *policy) {
+        JsonVariant *val;
+
+        assert(a);
+
+        val = security_assessor_find_in_policy(a, policy, "weight");
+        if  (val) {
+                    if (json_variant_type(val) == JSON_VARIANT_UNSIGNED)
```

Comment:
indentation off? should be multiple of 8 spaces.

Reply:
hmm, actually, it is better to use `json_variant_is_unsigned()` here. Why that? because json fields can have multiple types. i.e. there are only really numbers, and they can be used as floats, as unsigned or as signed integers. Thus, with a check like the above you only check for the "primary" type, even if the field could perfectly well qualify as the other too.